src/plugins/tools/magnifier/magnifierPanel.js

```diff
@@ -93,6 +103,39 @@ var dynamicComponentDefaultConfig = {
     left: 10
 };
 
+/**
+ * Identify scrolling dom nodes and store scrolling offsets
+ */
+const scrollingListenerCallback = _.throttle((event) => {
+    const $target = $(event.target);
+    const scrollingTop = event.target.scrollTop;
+    const scrollLeft = event.target.scrollLeft;
+    let scrollId = $target.data('magnifier-scroll');
+    let scrollData;
+
+    if (scrollId) {
+        //if the element is identified as a scrollable element, update it scrolling offset
+        scrollData = _.find(scrolling, {
+            id: scrollId
+        });
+        scrollData.scrollTop = scrollingTop;
+        scrollData.scrollLeft = scrollLeft;
+    } else {
+        //if the element is not yet identified as a scrollable element, tag it and register its id
+        scrollId = _.uniqueId('scrolling_');
+        $target.attr('data-magnifier-scroll', scrollId);
+        scrolling.push({
+            id: scrollId,
+            scrollTop: scrollingTop,
+            scrollLeft: scrollLeft
+        });
+    }
+
+    componentScrollCallback && componentScrollCallback(scrollData);
+}, scrollingDelay);
+
+window.addEventListener('scroll', scrollingListenerCallback, true);
```

Comment:
@jsconan  I am not sure about listening scrolls every time even user don't use the plugin.
Maybe there is some way to calculate scroll right before the plugin will show?

Reply:
Indeed, you are right. Even with the throttle applied on the callback, the event should only be applied when needed.

Reply:
yes, but i did not find another option how to restore scroll position when plugin is activated

Reply:
The issue is the scroll position is not accurate when the plugin is activated. Then why not properly initialize the callback context, without moving it?

Something like:
```
...
    let scrollPosition = getCurrentScrollPosition();
    var scrollingListenerCallback = _.throttle(function(event) {
        var $target = $(event.target);
        var scrollingTop = event.target.scrollTop + scrollPosition.top;
        var scrollLeft = event.target.scrollLeft + scrollPosition.left;
        ...
    }, scrollingDelay);
...
    /**
     * Initializes the listener for scrolling event and transfer the scrolling
     */
    function setScrollingListener() {
        scrollPosition = getCurrentScrollPosition();
        window.addEventListener('scroll', scrollingListenerCallback, true);
    }

    /**
     * Stops the listener for scrolling event
     */
    function removeScrollingListener() {
        window.removeEventListener('scroll', scrollingListenerCallback, true);
    }
...
```